graphql/resolvers.js

```
@@ -164,7 +175,23 @@ const resolvers = {
 
             if (!beacon) return new UserInputError("No beacon exists with that shortcode.");
             if (beacon.expiresAt < Date.now()) return new Error("Beacon has expired");
-            if (beacon.followers.includes(user)) return new Error("Already following the beacon");
+            for (let i = 0; i < beacon.followers.length; i++)
+                if (beacon.followers[i].id === user.id) {
+                    return new Error("Already following the beacon!");
+                }
+            if (beacon.leader === user.id) return new Error("Already leading the beacon!");
```

better to have this before the for loop so we can fail fast without hitting the loop

---

fixed!